athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/domain/predicate/expression/ConstantExpression.java

```diff
@@ -0,0 +1,89 @@
+/*-
+ * #%L
+ * Amazon Athena Query Federation SDK
+ * %%
+ * Copyright (C) 2019 - 2022 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connector.lambda.domain.predicate.expression;
+
+import com.amazonaws.athena.connector.lambda.data.Block;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.Collections.emptyList;
+
+public class ConstantExpression
+        extends FederationExpression
+{
+    public static final String DEFAULT_CONSTANT_EXPRESSION_BLOCK_NAME = "col1";
+    
+    private final Block valueBlock;
```

Comment:
So when I was writing this, I spent some time thinking about how to best represent the constants to keep it extensible since not all sources will be translating this into a string. TBH FederationExpressionParser should probably live in the athena-jdbc module.

So I wanted to keep the constant expression in its native value since it can really be any value, and there's no guarantee that all sources will just map it to string. That may be the case for JDBC, but its not guaranteed for all sources.

In addition, we already map constants to blocks within the ValueSets/summary object. The EquatableValueSet creates a new arrow block to represent constants and I wanted to keep things uniform across the framework.

The idea has always been that data should be converted from the Presto format to arrow to the datasource specific format, and I'm not sure if it makes sense to break that here.

Reply:
`FederationExpressionParser` can stay in athena-federation-sdk, bc it has an implementation `JdbcFederationExpressionParser` and in the future will have an `ElasticsearchFederationExpressionParser`, etc. Regardless of syntax, we'll want to keep the ability to recursively traverse the expressions in the sdk IMO.

Reply:
Sure traversing the expressions is helpful, but returning the entire object as a string is datasource specific.

Reply:
That's true. Ok for now, let me port it to athena-jdbc. In the future, we can make the traversal generic in the SDK and have connectors yield whatever is relevant for their client expression.

My change won't address anything about the ConstantExpression discussion you guys had above.